src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -154,10 +154,15 @@ public class PatientEverythingService : IPatientEverythingService
                         ? await SearchCompartment(resourceId, parentPatientId, since, types, encodedInternalContinuationToken, token, cancellationToken)
                         : await SearchCompartmentWithoutDate(resourceId, parentPatientId, since, types, encodedInternalContinuationToken, token, cancellationToken);
 
-                    if (!searchResult.Results.Any())
+                    // Starting from FHIR R5, "Devices" are included as part of Compartment Search.
+                    // Previous versions of FHIR should still query for "Devices" explicitly.
+                    if (ModelInfoProvider.Version < FhirSpecification.R5)
```

Comment:
Inject IModelInfoProvider is possible since its more declarative as a dependency